sycl/source/handler.cpp

```diff
@@ -93,6 +93,18 @@ event handler::finalize() {
     return MLastEvent;
   MIsFinalized = true;
 
+  auto setSubmissionTime = [&](detail::EventImplPtr event) {
+    if (!MQueue->is_host()) {
+      auto plugin = MQueue->getPlugin();
+      if (plugin.getBackend() == backend::ext_oneapi_level_zero) {
```

Comment:
I'm planning on having a separate PR where all the other backends are also implemented. For OpenCL [clGetDeviceAndHostTimer](https://man.opencl.org/clGetDeviceAndHostTimer.html) is only available for version 2.1 and later. Is there a way to the retrieve the API version for OpenCL?

Reply:
I am OK with staged approach but let's not add backend-specific code to SYCL RT, i.e. have plugins return something. 

> Is there a way to the retrieve the API version for OpenCL?

Sure, see here: https://github.com/intel/llvm/blob/516d41120cbf6996c66b3432352fe1806aef7934/sycl/plugins/opencl/pi_opencl.cpp#L606

Reply:
Thanks. I'm okay with the plugins return something, but I would still need backend-specific code as getting the submission time for the other backends is different than L0

Reply:
We need to define the API such that it works the same for all backends, and any difference in handling is within the plugins.

Reply:
In that case, I think it would be better just to combine whatever changes I was going to bring in the follow up PR into this one. As it would get rid of the backend specific code in SYCL RT.